ee/control/client_http.go

```diff
@@ -0,0 +1,154 @@
+package control
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"net/http"
+	"net/url"
+
+	"github.com/go-kit/kit/log"
+	"github.com/go-kit/kit/log/level"
+)
+
+// HTTPClient handles retrieving control data via HTTP
+type HTTPClient struct {
+	logger     log.Logger
+	addr       string
+	baseURL    *url.URL
+	client     *http.Client
+	insecure   bool
+	disableTLS bool
+}
+
+// controlRequest is the payload sent in control server requests
+type controlRequest struct {
+	// TODO: This is a temporary and simple data format for phase 1
+	Message string `json:"message"`
+}
+
+func NewControlHTTPClient(addr string, opts ...HTTPClientOption) (*HTTPClient, error) {
+	baseURL, err := url.Parse("https://" + addr)
```

Comment:
preference for Sprintf over concat
```suggestion
	baseURL, err := url.Parse(fmt.Sprintf("https://%s", "addr"))
```